File: app/src/main/java/bb/incognito/repositories/CocktailRepository.java
package bb.incognito.repositories;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.os.AsyncTask;

import java.util.List;

import bb.incognito.AppDatabase;
import bb.incognito.dao.CocktailDao;
import bb.incognito.model.Cocktail;

public class CocktailRepository {
    private CocktailDao cocktailDao;
    public LiveData<List<Cocktail>> getAllCocktails() { return allCocktails;}
    private LiveData<List<Cocktail>> allCocktails;

    public CocktailRepository(Application application, int guest_id) {
        AppDatabase db = AppDatabase.getDatabase(application);
        cocktailDao = db.cocktailDao();
    }

    public CocktailRepository(Application application){
        AppDatabase db = AppDatabase.getDatabase(application);
        cocktailDao = db.cocktailDao();
        allCocktails = cocktailDao.getAll();
    }

    public void insert(Cocktail cocktail) {
        new insertAsyncTask(cocktailDao).execute(cocktail);
    }

    private static class insertAsyncTask extends AsyncTask<Cocktail, Void, Void> {
        private CocktailDao cocktailDao;

        insertAsyncTask(CocktailDao guestDao) {
            this.cocktailDao = guestDao;
        }

        @Override
        protected Void doInBackground(final Cocktail... params) {
            //cocktailDao.insertCocktail(params[0]);
            return null;
        }
    }
}

File: app/src/main/java/bb/incognito/repositories/CocktailRepository.kt
package bb.incognito.repositories

import android.app.Application
import android.arch.lifecycle.LiveData
import android.os.AsyncTask

import java.util.ArrayList

import bb.incognito.AppDatabase
import bb.incognito.dao.CocktailDao
import bb.incognito.model.Cocktail
import bb.incognito.model.Guest

class CocktailRepository(application: Application) {
    private val cocktailDao: CocktailDao
    val allCocktails: LiveData<MutableList<Cocktail>>

    init {
        val db = AppDatabase.getDatabase(application)
        cocktailDao = db.cocktailDao()
        allCocktails = cocktailDao.all
    }

    fun insert(cocktail: Cocktail) {
        insertAsyncTask(cocktailDao).execute(cocktail)
    }

    private class insertAsyncTask internal constructor(private val cocktailDao: CocktailDao) : AsyncTask<Cocktail, Void, Void>() {

        override fun doInBackground(vararg params: Cocktail): Void? {
            cocktailDao.insertCocktail(params[0])
            return null
        }
    }
}
